### PR TITLE
Refine target_assign_op to unify the classification and regression targets assigning.

### PR DESCRIPTION
**Pull request by qingqing01**
Fix https://github.com/PaddlePaddle/Paddle/issues/8325

### CODE REVIEW

File: python/paddle/v2/fluid/tests/test_target_assign_op.py [hunk 1]
```diff
@@ -65,53 +65,88 @@ def target_assign(encoded_box, gt_label, match_indices, neg_indices, gt_lod,
...
-        trg_label_wt[i][neg_ids] = 1.0
+        if neg_indices is not None:
+            neg_ids = neg_indices[neg_lod[i]:neg_lod[i + 1]]
+            trg_label_wt[i][neg_ids] = 1.0
 
     return trg_box, trg_box_wt, trg_label, trg_label_wt
 
 
-class TestTargetAssginOp(OpTest):
+class TestTargetAssginOp1(OpTest):
```
Comment (wanghaox) [inline python/paddle/v2/fluid/tests/test_target_assign_op.py]: please use a more understandable name

File: python/paddle/v2/fluid/tests/test_target_assign_op.py [hunk 2]
```diff
@@ -65,53 +65,88 @@ def target_assign(encoded_box, gt_label, match_indices, neg_indices, gt_lod,
...
+        self.outputs = {
+            'Out': out,
+            'OutWeight': out_wt,
+        }
+
+    def test_check_output(self):
+        self.check_output()
+
+
+class TestTargetAssginOp2(OpTest):
```
Comment (wanghaox) [inline python/paddle/v2/fluid/tests/test_target_assign_op.py]: please use a more understandable name

File: python/paddle/v2/fluid/tests/test_target_assign_op.py (hunk 1 above)
Comment (qingqing01) [reply python/paddle/v2/fluid/tests/test_target_assign_op.py]: Done.

File: python/paddle/v2/fluid/tests/test_target_assign_op.py (hunk 2 above)
Comment (qingqing01) [reply python/paddle/v2/fluid/tests/test_target_assign_op.py]: Done.

**Pull request merged**